src/components/ContentSidebar/ContentSidebar.js

```diff
@@ -1005,12 +1046,53 @@ class ContentSidebar extends PureComponent<Props, State> {
         };
 
         if (SidebarUtils.canHaveSidebar(this.props)) {
-            this.api
-                .getTasksAPI(shouldDestroy)
-                .get(id, this.fetchTasksSuccessCallback, this.fetchTasksErrorCallback, requestData);
+            this.api.getTasksAPI(shouldDestroy).get({
+                id,
+                successCallback: this.fetchTaskAssignments,
+                errorCallback: this.fetchTasksErrorCallback,
+                params: requestData
+            });
         }
     }
 
+    /**
+     * Fetches the task assignments for each task
+     *
+     * @private
+     * @param {string} id - File id
+     * @return {void}
+     */
+    fetchTaskAssignments = (tasksWithoutAssignments: Tasks, shouldDestroy?: boolean = false): void => {
+        const { fileId }: Props = this.props;
+        if (!SidebarUtils.canHaveSidebar(this.props) || !fileId) {
+            return;
+        }
+
+        const requestData = {
+            params: {
+                fields: TASK_ASSIGNMENTS_FIELDS_TO_FETCH.toString()
+            }
+        };
+
+        let tasks = tasksWithoutAssignments;
+        const { entries } = tasksWithoutAssignments;
+        const taskAssignmentPromises = [];
+        entries.forEach((task) => {
+            const promise = this.api.getTasksAPI(shouldDestroy).getAssignments(
```

Comment:
get doesn't return a promise, how does this work exactly?

Reply:
@jeremypress  If you'll notice, Sumedha has modified it to return a promise

Reply:
`get()` isn't it's own async method. whereas `makeRequest()` is. I modified `get()` to return the promise from `makeRequest()`